### PR TITLE
Fix examples build

### PR DESCRIPTION
**Pull request by glibsm**
Examples makefile had the wrong syntax for the default target
and wasn't building the server component.

After fixing that, there was a small chain reaction with examples
not bothering to update the system thriftrw.

When running uberfx `make test`, run through full examples
makefile, rather than codegen to find failures early.

### CODE REVIEW

File: examples/keyvalue/Makefile [hunk 1]
```diff
@@ -7,4 +9,7 @@ server/server: $(wildcard server/*.go) kv/types.go
 .PHONY: all
 all: server/server
 
-.DEFAULT_GOAL: all
+.PHONY: deps
+deps:
+	$(ECHO_V)go get -v go.uber.org/thriftrw
```
Comment (anuptalwalkar) [inline examples/keyvalue/Makefile]: Wait. I though we were installing from ../../vendor/ and not do go get? This will lead to failures once yarpc update happens.

File: .build/deps.mk [hunk 2]
```diff
@@ -4,8 +4,8 @@ libdeps:
 	$(ECHO_V)glide --version 2>/dev/null ||	go get -v github.com/Masterminds/glide
 	$(ECHO_V)glide install
 
-.PHONY: dependencies
-dependencies: libdeps
+.PHONY: deps
```
Comment (sectioneight) [inline .build/deps.mk]: why this change?

File: examples/keyvalue/Makefile [hunk 3]
```diff
@@ -7,4 +9,7 @@ server/server: $(wildcard server/*.go) kv/types.go
...
+	$(ECHO_V)go get -v go.uber.org/yarpc/encoding/thrift/thriftrw-plugin-yarpc
```
Comment (sectioneight) [inline examples/keyvalue/Makefile]: This is wrong. We need to use the version that matches what we have in glide.yaml, otherwise the type signatures are going to mismatch. The YARPC team noticed and fixed this for us while you were away in december.

File: examples/keyvalue/Makefile (hunk 1 above)
Comment (glibsm) [inline examples/keyvalue/Makefile]: yarpc is stable now, no?

File: .build/deps.mk (hunk 2 above)
Comment (glibsm) [inline .build/deps.mk]: `make deps` is in examples, and `make dependencies` is annoying to type, especially since it doesn't autocomplete from the included .mk file :(

File: examples/keyvalue/Makefile (hunk 3 above)
Comment (glibsm) [inline examples/keyvalue/Makefile]: ooookaaaay. @anuptalwalkar made the same point

File: examples/keyvalue/Makefile (hunk 1 above)
Comment (sectioneight) [inline examples/keyvalue/Makefile]: It is now, but what happens when 2.0 comes out and we're using 1.0 in glide, but this fetches the latest (2.0)?

File: glide.yaml [hunk 4]
```diff
@@ -26,6 +26,9 @@ import:
   subpackages:
   - assert
   - require
+- package: github.com/golang/mock
```
Comment (anuptalwalkar) [inline glide.yaml]: Does this mean anyone who needs thriftrw, will also need mock package?

Comment (glibsm) [inline glide.yaml]: By the looks of it, yes. It doesn't work locally if you remove the `mock` package from system path. 

Looks like just including thriftrw doesn't pull in the mock in the lockfile.

/cc @breerly

**Comment by sectioneight** [comment]
We shouldn't need mocks in production code, right?

File: glide.yaml (hunk 4 above)
Comment (anuptalwalkar) [inline glide.yaml]: yikes. I was hoping glide up would pull the package. Oh well.

**Comment by glibsm** [comment]
@sectioneight thrift codegen uses mocks, so yeah we need it in production by the looks of it. Unless something is wrong with our yarpc setup. I cc'd @breerly here so we can comment as well.

/bcc @abhinav

**Comment by abhinav** [comment]
Starting 1.1, the YARPC ThriftRW plugin generates gomock mocks by default in a
separate subpackage (`${serviceName}test`). To disable this, you can use the
`-no-gomock` flag on the plugin.

    thriftrw -p 'yarpc -no-gomock' foo.thrift

Generally, it shouldn't be a problem to have that code in your repo though.
You don't actually have a dependency on `gomock` until you use that package.
`glide` won't even add the dependency to your glide.lock unless you're
actually importing the test package.

**Comment by sectioneight** [comment]
Right, so we shouldn't be having to add gomock as an explicit dependency,
right?

**Comment by abhinav** [comment]
Sounds right.

File: glide.yaml (hunk 4 above)
Comment (sectioneight) [inline glide.yaml]: According to the convo with @abhinav this shouldn't be necessary. Please convince me it is. I don't like this.
